dask/dataframe/io/parquet/arrow.py

```diff
@@ -745,24 +814,57 @@ def read_partition(
         if not isinstance(row_group, list):
             row_group = [row_group]
 
-        dfs = []
+        tables = []
         for rg in row_group:
-            piece = pq.ParquetDatasetPiece(
-                path,
-                row_group=rg,
-                partition_keys=partition_keys,
-                open_file_func=partial(fs.open, mode="rb"),
-            )
-            arrow_table = cls._parquet_piece_as_arrow(
-                piece, columns, partitions, **kwargs
-            )
-            df = cls._arrow_table_to_pandas(arrow_table, categories, **kwargs)
-
-            if len(row_group) > 1:
-                dfs.append(df)
+            if pa_ds is not None and isinstance(rg, pa_ds.ParquetFileFragment):
+                # `rg` is already a `ParquetFileFragment`, pyarrow
+                # knows how to convert this to a `table`
+                cols = []
+                for name in columns:
+                    if name is None:
+                        if "__index_level_0__" in schema.names:
+                            columns.append("__index_level_0__")
+                    else:
+                        cols.append(name)
+                arrow_table = rg.to_table(
+                    use_threads=False,
+                    schema=schema,
+                    columns=cols,
+                    filter=pq._filters_to_expression(filters) if filters else None,
+                )
+            else:
+                piece = pq.ParquetDatasetPiece(
```

Comment:
Also, we should still be able to handle the case where row_group is a list of integers ids also for the new dataset API? Because now the new ArrowDatasetEngine is still using ParquetDatasetPiece?

Reply:
>could you use read_table/ ParquetFile instead of of ParquetDatasetPiece?

Good point - I made some changes to allow `ParquetFile.read_row_groups`/`read`.

>Also, we should still be able to handle the case where row_group is a list of integers ids also for the new dataset API? Because now the new ArrowDatasetEngine is still using ParquetDatasetPiece?

At first, I was only using the `Fragment.to_table` approach in `ArrowDatasetEngine`.  However, I was testing performance with a pathological case (1,440 small row-groups; 1000 rows each), and noticed that the `ParquetDatasetPiece` approach (and `ParquetFile`) was significantly faster.  For a smaller number of large row groups, the fragment-based approach performs quite well, so I am not sure if this is just an issue with row-group size.

Regardless of the performance issue, it would certainly be great to have the option to read a subset of row-groups from a single `FileFragment` in one go (to avoid the need for concatenation).

Reply:
> However, I was testing performance with a pathological case (1,440 small row-groups; 1000 rows each), and noticed that the ParquetDatasetPiece approach (and ParquetFile) was significantly faster.

Was that for reading as a single partition, or with `split_row_groups=True` ?  
(or if you have some sample code of what you were testing, that would certainly be interesting to see if there is some overhead to optimize for the fragment code)